method/publish.json

```diff
@@ -0,0 +1,56 @@
+{
+	"$schema": "http://json-schema.org/draft/2019-09/schema#",
+	"$id": "http://pop2020.dedis.ch/publish.json",
+	"description": "Match publish query",
+	"type": "object",
+
+	"properties": {
+		"method": { 
+			"description": "[String] operation to be performed by the query",
+			"const": "publish" 
+		},
+		"params": {
+			"oneOf": [
+				{
+					"description": "Match a publish query destined for the root channel",
+					"type": "object",
+					"properties": {
+						"channel": { 
+							"description": "[String] name of the (root) channel", 
+							"$ref": "channel/rootChannel.json",
+							"$comment": "Note: should always be \"/root\""
+						},
+						"message": { 
+							"description": "[Message] message to be published",
+							"$ref": "message/messageGeneral.json"
+						}
+					},
+					"additionalProperties": false,
+					"required": ["channel", "message"]
+				}, 
+				{
+					"description": "Match a publish query destined for a sub-channel",
+					"type": "object",
+					"properties": {
+						"channel": { 
+							"description": "[String] name of the channel", 
+							"$ref": "channel/subChannel.json"
+						},
+						"message": { 
+							"description": "[Message] message to be published",
+							"oneOf": [
+								{ "$ref": "message/messageGeneral.json" },
+								{ "$ref": "message/messageWitnessMessage.json" }

```

Comment:
I may be missing something, but isn't the witness message a sort of sub-type of messageGeneral? 
If so, there's no need for the one-of.

Reply:
Yes! Now I remember why I did it! This is tied to your next comment (just below). If we post on the root channel, then the message must be a `messageGeneral` (and thus not a `messageWitnessMessage`). The difference between them it that the witness signature list is always empty for a `messageWitnessMessage`. 

Since messages (`messageGeneral` & `messageWitnessMessage`) and methods (`publish`, `catchup`, ...) are at different levels in the tree, there's no way for me to check that a message sent on the root channel is indeed *not* a `messageWitnessMessage`